### PR TITLE
Fix poms to contain versions

### PR DESCRIPTION
**Pull request by j-baker**
Does a few things, because stuff kept breaking.

1. Upgrade gradle.
2. Upgrade nebula and other plugins.
3. Upgrade publishing to use the same setup as http-remoting-api since I was having issues with gradle.
4. Upgrade baseline
5. Make compatible with new baseline

### CODE REVIEW

**Comment by j-baker** [comment]
@uschi2000 do I need to do anything with bintray creds here? e.g. do I need different ones or will the current ones be good?

File: encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java [hunk 1]
```diff
@@ -30,28 +29,35 @@
  * algorithm with a supported key.
  */
 public enum Algorithm {
-    AES("AES", () -> AesKeyPair.newKeyPair(), () -> AesGcmEncrypter.INSTANCE),
-    RSA("RSA", () -> RsaKeyPair.newKeyPair(), () -> RsaOaepEncrypter.INSTANCE);
+    AES("AES", AesGcmEncrypter.INSTANCE) {
+        @Override
+        public KeyPair newKeyPair() {
```
Comment (j-baker) [inline encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java:34]: errorprone demands this?

https://github.com/google/error-prone/blob/master/docs/bugpattern/ImmutableEnumChecker.md

File: encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java [hunk 2]
```diff
@@ -30,28 +29,35 @@
  * algorithm with a supported key.
  */
 public enum Algorithm {
-    AES("AES", () -> AesKeyPair.newKeyPair(), () -> AesGcmEncrypter.INSTANCE),
-    RSA("RSA", () -> RsaKeyPair.newKeyPair(), () -> RsaOaepEncrypter.INSTANCE);
+    AES("AES", AesGcmEncrypter.INSTANCE) {
```
Comment (bavardage) [inline encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java:32]: why these changes?

File: encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java [hunk 3]
```diff
@@ -30,28 +29,35 @@
...
+        }
+    };
 
     private final String name;
-    private final Supplier<KeyPair> keyPairGenerator;
-    private final Supplier<Encrypter> encrypterGenerator;
+    private final Encrypter encrypter;
 
-    Algorithm(String name, Supplier<KeyPair> keyPairGenerator, Supplier<Encrypter> cipherGenerator) {
+    Algorithm(String name, Encrypter cipher) {
```
Comment (bavardage) [inline encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java:48]: why this change?

Comment (j-baker) [reply encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java:48]: see above - errorprone was demanding it

Comment (bavardage) [reply encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java:48]: fair enough

**Comment by uschi2000** [comment]
current creds should be good

File: encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java (hunk 1 above)
Comment (bavardage) [reply encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java:34]: ok so we were being naughty with our enums before by having them take suppliers I guess

**Review by bavardage — APPROVED** [review]
